### ISSUE TITLE
IMU calibration parameters in ARCORE

### ISSUE THREAD

**Issue by Momeo** [google-ar/arcore-android-sdk#973]
We notice that some Android devices (ARCORE supported) don’t work well in 3D tracking because of the low accuracy of calibration. We need to make the tracking more accurate in our project for business customer. 

Is it possible for Google to provide the API for us to set the IMU calibration parameters in the future?

We assume the reason of some devices occurs zero drift problems is that Google only gathered the parameters of few devices and just average the results to calibrating the IMU parameters.

**Comment by lvonasek** [comment]
I would also appreciate this feature. ARCore's inaccuracy makes impossible to make my app get out of beta.

**Comment by andrestone** [comment]
I'm also facing this same problem, but after reading this comment by **inio** from 2018, I've started searching another path to the solution for this problem.

> Calibration means a lot more than just IMU calibration. Calibration comprises:
IMU calibration (biases, scale factors, covariance, etc.). This often change for each model number of a phone.
Camera intrinsics calibration (pinhole model, distortion parameters, rolling shutter time). Similarly, this often chagnes every time any part of the camera changes (sensor, optics, focus mechanism, camera driver software).
Camera extrinsics calibration (position/orientation relative to IMU). Any change to the location or orientation of the camera or IMU within the device will require a recalibration.
Relative timestamping behavior of IMU & Camera. This is assuming we even get consistent, precise timestamps for both, which can often completely prevent enabling a device. Even OS updates with no apparent change to the camera or sensor subsystems can alter this.
The camera (and sometimes IMU) configuration required to get the above. We need to find a camera configuration that will deliver consistent rolling shutter time, consistent timestamping, full-sensor output for motion tracking, cropped texture output for display, fixed focus and reasonable auto-exposure behavior.
Finally, we can't do the above on just one device. We need to calibrate across many units of a device, taking an average and them and then validating that calibration on a completely separate set of many more units. We need to be careful acquiring these devices to ensure we cover a wide range of build dates and build sites.
TL;DR: There's a lot more than just IMU calibration required to enable a device for ARCore.

**Comment by Momeo** [comment]
> I'm also facing this same problem, but after reading this comment by **inio** from 2018, I've started searching another path to the solution for this problem.

Actually we have so many ways to calibrate IMU for each device to achieve a better performance, I wonder why there is no API to set it.
